source/Mlos.Python/mlos/Optimizers/ExperimentDesigner/UtilityFunctionOptimizers/RandomSearchOptimizer.py

```diff
@@ -43,21 +43,32 @@ def __init__(
         UtilityFunctionOptimizer.__init__(self, optimizer_config, optimization_problem, utility_function, logger)
 
     @trace()
-    def suggest(self, context_values_dataframe: pd.DataFrame = None):  # pylint: disable=unused-argument
-        """ Returns the next best configuration to try.
+    def maximize(self, target_function, context_values_dataframe=None):
+        """Maximize callable target function.
 
-        It does so by generating num_samples_per_iteration random configurations,
-        passing them through the utility function and selecting the configuration with
-        the highest utility value.
+        Parameters
+        ----------
+        target_function : callable
+            Function to maximize.
+        context_values_dataframe : DataFrame (default=None)
+            Context for optimization.
 
-        TODO: make it capable of consuming the context values
-        :return:
+        Returns
+        -------
+        position_of_optimum : Point
         """
+        config_values_dataframe = self.optimization_problem.parameter_space.random_dataframe(num_samples=self.optimizer_config.num_samples_per_iteration)
+        if context_values_dataframe is not None:
+            assert len(context_values_dataframe) == 1
+            config_values_dataframe['_join_key'] = 0
+            copied_context = context_values_dataframe.copy()

```

Comment:
This exact same logic is needed for GlowWormSwarmOptimizer and every future utility function optimizer we write, no?

Reply:
oh sorry, I thought It had put it in the parent class, will fix.